trans.py

```diff
@@ -16,80 +35,85 @@ def find_header(counter, width):
-    logging.info("Position of last line, %s", LINE_COUNT)
-    logging.info("%s bytes in header",IN_POS)
-    if IN_POS == -1 or LINE_COUNT == -1:
-        logging.error("End of header not found")
-
-    # To read the information to legible format
-    typeInfo, bytePack, varNames = read_pack("GNV1B")
-    logging.info(typeInfo, bytePack, varNames)
-    logging.info("Initial position, %s",IN_POS)
-    packTotal = sum(bytePack)  #Total length of pack. in bytes
-    logging.info("The length of a pack. is %s",packTotal)
-    PACK_LENGTH = len(bytePack)
-    # For debugging purposes
-    if len(bytePack) == len(typeInfo):
-        logging.info('There is an information type for each set of bytes')
-    else:
-        logging.warning('There are leftover packs of data without a type of information')
-    # To check if there are leftover bytes
-    leftoverBytes = (len(lines)- IN_POS)%packTotal
-    logging.warning('There are %s leftover bytes',leftoverBytes)
-    if leftoverBytes:
-        IN_POS += leftoverBytes     # So the initial position is adequate to finish in a block
-    logging.warning('The initial position has been switched to %s', IN_POS)
-    # To interpret the information
-    decodPack = []
-    while IN_POS < len(lines):  # Iterate through the read file
-        IC = 0
-        for byte in bytePack:   # Iterate through one data pack
-            if typeInfo[IC] == "int":
-                decodPack.append(struct.unpack('>I',lines[IN_POS:(IN_POS+byte)]))
-            elif typeInfo[IC] == "chr":
-                decodPack.append((lines[IN_POS:(IN_POS+byte)]).decode("ascii"))
-            elif typeInfo[IC] == "dp":
-                decodPack.append(struct.unpack('>d',lines[IN_POS:(IN_POS+byte)]))
-            elif typeInfo[IC] == "uchar":
-                decodPack.append((lines[IN_POS:(IN_POS+byte)].decode("ascii")))
-            IN_POS += byte
-            IC += 1
-    print(decodPack[0:160])
+fileNames, numFiles = read_file()
+typeInfo, bytePack, varNames = read_pack(fileNames) # getting the information to know what to read
+decodPack = [[]  for x in range(numFiles)] # empty list of lists to store results
+print(fileNames)
+for _n in range(numFiles): # to iterate through each set of data
+    with open(fileNames[_n], "rb", encoding=None) as input_file:
+        # To ascertain the position of the header
+        lines = input_file.read()
+        NEWLINE_CHAR = 0x0A  # from HEX reader for the file GNV
+        header_width = lines.find(NEWLINE_CHAR)
+        logging.info('Header width is %s', header_width)
+        LINE_COUNT = 0
+        firstline = lines[0:header_width].decode("ascii")
+        print(firstline)
+        IN_POS, LINE_COUNT = find_header(LINE_COUNT, header_width)
+        logging.info("Position of last line, %s", LINE_COUNT)
+        logging.info("%s bytes in header",IN_POS)
+        if IN_POS == -1 or LINE_COUNT == -1:
+            logging.error("End of header not found")
```

Comment:
If you log an error maybe you want to configure a user defined exception https://www.scaler.com/topics/user-defined-exception-in-python/